---
title: "COMET differential expression"
output:
  html_notebook:
    toc: yes
    toc_float: yes
editor_options:
  chunk_output_type: console
---

```{r setup, include=FALSE} 
knitr::opts_chunk$set(echo = FALSE)
knitr::opts_chunk$set(message = FALSE)
knitr::opts_chunk$set(warning = FALSE)
```

```{r}
setwd("~/Box/COVIDvsARDS/")
rm(list = ls())
library(tidyverse)
library(readxl)
library(magrittr)
library(ggpubr)
library(DESeq2)
library(BiocParallel)
register(MulticoreParam(2))
library(tableone)
library(pheatmap)
library(scales)
library(IHW)
source("~/Box/rnaseqscripts/rnaseqscripts.R")

load("~/Box/data/comet/clean/comet.genes.Rda")

#Data frame to link ENSG ids with HGNC symbols
ensembltohgnc <- comet.genes %>% dplyr::select(ensembl_gene_id,gene_symbol) %>% dplyr::rename(hgnc_symbol = gene_symbol) %>% as.data.table

comet.genes %<>% column_to_rownames("ensembl_gene_id") %>% dplyr::select(-gene_symbol)
```

```{r themes}
canonical.theme <- list(
    theme_classic(),
    theme(axis.text.y = element_text(size = 10),
          axis.title.y = element_blank(),
          legend.position = "blank"),
    scale_fill_identity()
    )

col.covid <- "firebrick3"
col.clear <- "grey90"
col.ards <- "mediumorchid3"
col.viral <- "dodgerblue2"
col.bacterial <- "springgreen3"

ann_colors = list(
  LRTItype = c("vir" = "#00FF00", "bac" = "#0000FF", "vir+bac" = "#00FFFF", "unk" = "#AAAAAA", "-" = "#AAAAAA", "no" = "#000000"),
  study = c("MBAL" = "#87ceeb", "COMET" = "#ff8c69"),
  dx = c("ARDS" = col.ards, "COVID" = col.covid, "Clear" = col.clear, "ViralARDS" = col.viral, "BacterialARDS" = col.bacterial)
)

heatmaprows <- 35
```

## COVID-19 vs. other causes of ARDS

```{r}
exp1.samples <- read_csv("./covidardsmetadata-final.csv")
colnames(exp1.samples) %>% cat(sep = "', '")
tab1vars <- c('Age', 'ARDS etiology', 'Gender', 'Race', 'Hispanic Ethnicity', '30-day mortality', 'LRTI Type', 'Minimum PF Ratio')

exp1.samples %<>%
  mutate(dx = case_when(
    `ARDS group` == "COVID-ARDS" ~ "COVID",
    `ARDS group` == "Other-ARDS" ~ "ARDS",
    `ARDS group` == "No-ARDS" ~ "Clear"
  )) %>% 
  mutate(dx = factor(dx, levels = c("COVID", "ARDS", "Clear"))) %>%
  mutate(`Minimum PF Ratio` = as.numeric(`Minimum PF Ratio`)) %>%
  dplyr::select(-`ARDS group`)


exp1.samples %>%
  filter(dx %in% c("COVID", "ARDS")) %>%
  mutate(dx = factor(dx, levels = c("COVID", "ARDS"))) %>%
CreateTableOne(data = ., 
                              strata ="dx",
                              vars = tab1vars,
                              test = T
                              ) %>% print(nonnormal = c("Minimum PF", "Age"))

exp1.samples %>%
  filter(dx %in% c("COVID", "Clear")) %>%
  mutate(dx = factor(dx, levels = c("COVID", "Clear"))) %>%
CreateTableOne(data = ., 
                              strata ="dx",
                              vars = tab1vars,
                              test = T
                              ) %>% print(nonnormal = c("Minimum PF", "Age"))

exp1.samples %<>%
  dplyr::rename(
    sampleid = `RNASeq ID`, 
    lrtitype = `LRTI Type`
  )


exp1.genes <-
  comet.genes %>% 
  dplyr::select(exp1.samples$sampleid)
```

```{r message=FALSE, warning=FALSE}
exp1.dds <- 
  DESeqDataSetFromMatrix(
    countData = exp1.genes,
    colData = exp1.samples,
    design = ~ dx
  )

exp1.dds <- DESeq(exp1.dds, parallel = T)

#This function extracts the results for each contrast, shrinks using ashr, and saves the results in a folder as a CSV file and as an object. 
deseqresults(exp1.dds, method = "ashr")
```



## COVID-19 vs. bacterial or viral ARDS
```{r}
exp2.samples <- exp1.samples %>%
  filter((dx == "ARDS" & (lrtitype== "Viral" | lrtitype == "Bacterial")) |
           (dx == "COVID" & lrtitype == "Viral")|
           dx == "Clear") %>%
  mutate(dx = case_when(
    dx == "COVID" ~ "COVID", 
    dx == "Clear" ~ "Clear",
    lrtitype == "Viral" ~ "ViralARDS", 
    lrtitype == "Bacterial" ~ "BacterialARDS" 
  )) %>%
  mutate(dx = factor(dx, levels = c("COVID", "BacterialARDS", "ViralARDS", "Clear")))
```

```{r message=FALSE, warning=FALSE}
exp2.genes <-
  comet.genes %>% dplyr::select(one_of(exp2.samples$sampleid))

exp2.dds <- 
  DESeqDataSetFromMatrix(
    countData = exp2.genes,
    colData = exp2.samples,
    design = ~ dx
  )

exp2.dds <- DESeq(exp2.dds, parallel = T)
deseqresults(exp2.dds, methods = "ashr")
```

### Heatmaps
```{r exp1heatmap, fig.cap = "Heatmap of differentially expressed genes (FDR0.1) for COVID vs. ARDS", fig.height = 16, fig.width = 10}
exp1.vst <- vst(exp1.dds, blind = TRUE)
exp1.nm <- exp1.vst %>% assay 
exp1.nm %<>% as.data.frame%>% rownames_to_column("ensembl_gene_id") %>% left_join(ensembltohgnc) %>% distinct(hgnc_symbol, .keep_all = T) %>% filter(hgnc_symbol != "") %>% column_to_rownames("hgnc_symbol") %>% dplyr::select(-ensembl_gene_id)

ARDSvsCOVID.DEgenes <- 
  exp1.ARDSvsCOVID.ashr %>%
  filter(ihwP < 0.1) %>% 
  mutate(direction = sign(log2FoldChange)) %>%
  group_by(direction) %>%
  arrange(pvalue, .by_group = T) %>%
  filter(row_number() <= heatmaprows) %>%
  ungroup %>%
  dplyr::select(hgnc_symbol) %>%
  unlist

ARDSvsCOVID.nm  <-  
  exp1.nm %>%
  filter(rownames(.) %in% ARDSvsCOVID.DEgenes) %>%
  t %>%
  as.data.frame %>%
  scale

exp1.breaks <- quantile_breaks(ARDSvsCOVID.nm %>% t)
exp1.annotation <- exp1.samples %>%
  dplyr::select(sampleid, dx) %>%
  column_to_rownames("sampleid")

pheatmap.rotate(
  ARDSvsCOVID.nm %>% t,
  show_rownames = T,
  show_colnames = F,
  scale = "row",
  cutree_rows = 2,
  cutree_cols = 1,
  border_color = NA,
  breaks = exp1.breaks,
  color = viridis(9),
  fontsize_col = 10,
  fontsize_row = 8,
  cluster_rows = T,
  cluster_col = T,
  treeheight_row = 20,
  treeheight_col = 20,
  annotation_legend = TRUE,
  annotation_colors = ann_colors,
  annotation_col = exp1.annotation,
  clustering_distance_cols = "euclidean",
  clustering_distance_rows = "manhattan",
  uselastdend = F,
  rotate = F, 
  cellwidth = 10,
  cellheight = 10
  )
```

```{r}
exp2.vst <- vst(exp2.dds, blind = TRUE)
exp2.nm <- exp2.vst %>% assay 
exp2.nm %<>% as.data.frame%>% rownames_to_column("ensembl_gene_id") %>% left_join(ensembltohgnc) %>% distinct(hgnc_symbol, .keep_all = T) %>% filter(hgnc_symbol != "") %>% column_to_rownames("hgnc_symbol") %>% dplyr::select(-ensembl_gene_id)
```


```{r, fig.height = 16, fig.width = 10}
BacterialARDSvsCOVID.DEgenes <-   exp2.BacterialARDSvsCOVID.ashr %>%
  filter(ihwP < 0.1) %>% 
  mutate(direction = sign(log2FoldChange)) %>%
  group_by(direction) %>%
  arrange(pvalue, .by_group = T) %>%
  filter(row_number() <= heatmaprows) %>%
  ungroup %>%
  dplyr::select(hgnc_symbol) %>%
  unlist

BacterialARDSvsCOVID.samples <- 
  exp2.samples %>%
  filter(dx %in% c("COVID", "BacterialARDS")) %>% 
  dplyr::select(sampleid) %>%
  unlist

BacterialARDSvsCOVID.nm  <- 
  exp2.nm %>%
  filter(rownames(.) %in% BacterialARDSvsCOVID.DEgenes) %>%
  t %>%
  as.data.frame %>% 
  filter(rownames(.) %in% BacterialARDSvsCOVID.samples) %>%
  scale


exp2.breaks <- quantile_breaks(BacterialARDSvsCOVID.nm %>% t)
exp2.annotation <- exp2.samples %>%
  dplyr::select(sampleid, dx) %>%
  column_to_rownames("sampleid")

pheatmap.rotate(
  BacterialARDSvsCOVID.nm %>% t,
  show_rownames = T,
  show_colnames = F,
  scale = "row",
  cutree_rows = 2,
  cutree_cols = 1,
  border_color = NA,
  breaks = exp2.breaks,
  color = viridis(9),
  fontsize_col = 10,
  fontsize_row = 8,
  cluster_rows = T,
  cluster_col = T,
  treeheight_row = 20,
  treeheight_col = 20,
  annotation_legend = TRUE,
  annotation_colors = ann_colors,
  annotation_col = exp2.annotation,
  clustering_distance_cols = "euclidean",
  clustering_distance_rows = "manhattan",
  uselastdend = F,
  rotate = F, 
  cellwidth = 10,
  cellheight = 10
  )
```


```{r, fig.height = 16, fig.width = 10}
ViralARDSvsCOVID.DEgenes <- 
  exp2.ViralARDSvsCOVID.ashr %>%
  filter(ihwP < 0.1) %>% 
  mutate(direction = sign(log2FoldChange)) %>%
  group_by(direction) %>%
  arrange(pvalue, .by_group = T) %>%
  filter(row_number() <= heatmaprows) %>%
  ungroup %>%
  dplyr::select(hgnc_symbol) %>%
  unlist

ViralARDSvsCOVID.samples <- 
  exp2.samples %>%
  filter(dx %in% c("COVID", "ViralARDS")) %>% 
  dplyr::select(sampleid) %>%
  unlist

ViralARDSvsCOVID.nm  <- 
  exp2.nm %>%
  filter(rownames(.) %in% ViralARDSvsCOVID.DEgenes) %>%
  t %>%
  as.data.frame %>% 
  filter(rownames(.) %in% ViralARDSvsCOVID.samples) %>%
  scale

pheatmap.rotate(
  ViralARDSvsCOVID.nm %>% t,
  show_rownames = T,
  show_colnames = F,
  scale = "row",
  cutree_rows = 2,
  cutree_cols = 1,
  border_color = NA,
  breaks = exp2.breaks,
  color = viridis(9),
  fontsize_col = 10,
  fontsize_row = 8,
  cluster_rows = T,
  cluster_col = T,
  treeheight_row = 20,
  treeheight_col = 20,
  annotation_legend = TRUE,
  annotation_colors = ann_colors,
  annotation_col = exp2.annotation,
  clustering_distance_cols = "euclidean",
  clustering_distance_rows = "manhattan",
  uselastdend = F,
  rotate = F, 
  cellwidth = 10,
  cellheight = 10
  )
```
