---
title: "COMET differential expression"
output:
  html_document:
    toc: yes
    df_print: paged
  html_notebook:
    toc: yes
    toc_float: yes
editor_options:
  chunk_output_type: console
---

```{r knitrsetup, include=FALSE} 
rm(list = ls())
knitr::opts_chunk$set(echo = FALSE)
knitr::opts_chunk$set(message = FALSE)
knitr::opts_chunk$set(warning = FALSE)
```

```{r options}
#Use this section to set options
working.directory <- "~/Box/COVIDvsARDS/Files for GitHub/"
```

```{r setup}
setwd(working.directory)
library(tidyverse)
library(readxl)
library(magrittr)
library(ggpubr)
library(DESeq2)
library(BiocParallel)
register(MulticoreParam(2))
library(tableone)
library(pheatmap)
library(scales)
library(IHW)
source("rnaseqfunctions.r")
```

```{r themes}
col.covid <- "firebrick3"
col.clear <- "grey90"
col.ards <- "mediumorchid3"
col.viral <- "dodgerblue2"
col.bacterial <- "springgreen3"

ann_colors = list(
  LRTItype = c("vir" = "#00FF00", "bac" = "#0000FF", "vir+bac" = "#00FFFF", "unk" = "#AAAAAA", "-" = "#AAAAAA", "no" = "#000000"),
  study = c("MBAL" = "#87ceeb", "COMET" = "#ff8c69"),
  dx = c("ARDS" = col.ards, "COVID" = col.covid, "Clear" = col.clear, "ViralARDS" = col.viral, "BacterialARDS" = col.bacterial)
)

heatmaprows <- 50
```

## COVID-19 vs. other causes of ARDS

```{r loaddata}
exp1.genes <- read_csv("./comet.genes.csv")

#Data frame to link ENSG ids with HGNC symbols
ensembltohgnc <- exp1.genes %>% dplyr::select(ensembl_gene_id,gene_symbol) %>% dplyr::rename(hgnc_symbol = gene_symbol)

exp1.genes %<>% column_to_rownames("ensembl_gene_id") %>% dplyr::select(-gene_symbol)

exp1.samples <- read_csv("./covidardsmetadata-final.csv")



exp1.samples %<>%
  mutate(dx = case_when(
    `ARDS group` == "COVID-ARDS" ~ "COVID",
    `ARDS group` == "Other-ARDS" ~ "ARDS",
    `ARDS group` == "No-ARDS" ~ "Clear"
  )) %>% 
  mutate(dx = factor(dx, levels = c("COVID", "ARDS", "Clear"))) %>% #Set COVID as the reference value, so that all DESeq contrasts are with COVID
  mutate(`Minimum PF Ratio` = as.numeric(`Minimum PF Ratio`)) %>%
  dplyr::select(-`ARDS group`)

#Variables from metadata table to use for Table 1
tab1vars <- c('Age', 'ARDS etiology', 'Gender', 'Race', 'Hispanic Ethnicity', '30-day mortality', 'LRTI Type', 'Minimum PF Ratio')

#Calculate summary statistics for included COVID and OtherARDS subjects
exp1.samples %>%
  filter(dx %in% c("COVID", "ARDS")) %>%
  mutate(dx = factor(dx, levels = c("COVID", "ARDS"))) %>%
CreateTableOne(data = ., 
                              strata ="dx",
                              vars = tab1vars,
                              test = T
                              ) %>% print(nonnormal = c("Minimum PF", "Age"))

#Calculate summary statistics for included COVID and NoARDS subjects
exp1.samples %>%
  filter(dx %in% c("COVID", "Clear")) %>%
  mutate(dx = factor(dx, levels = c("COVID", "Clear"))) %>%
CreateTableOne(data = ., 
                              strata ="dx",
                              vars = tab1vars,
                              test = T
                              ) %>% print(nonnormal = c("Minimum PF", "Age"))


exp1.samples %<>%
  dplyr::rename(
    sampleid = `RNASeq ID`, 
    lrtitype = `LRTI Type`
  )
```

```{r message=FALSE, warning=FALSE}
exp1.dds <- 
  DESeqDataSetFromMatrix(
    countData = exp1.genes,
    colData = exp1.samples,
    design = ~ dx
  )

exp1.dds <- DESeq(exp1.dds, parallel = T)

#This function extracts the results for each contrast in a DESeq object and shrinks log2-fold changes using ashr or apeglm. In addition, the function uses the IHW package to calculate false detection rates. The results for each contrast are saved in an object in the global environment. The results are also exported as a CSV file in a subfolder of the active working directory. 
deseqresults(exp1.dds, method = "ashr")
```



## COVID-19 vs. bacterial or viral ARDS
```{r message=FALSE, warning=FALSE}
#Select the subjects with secondary analysis
exp2.samples <- exp1.samples %>%
  filter((dx == "ARDS" & (lrtitype== "Viral" | lrtitype == "Bacterial")) | #Subjects with viral or bacterial ARDS
           (dx == "COVID" & lrtitype == "Viral")| #COVID subjects without co-infections
           dx == "Clear") %>% #NoARDS controls
  mutate(dx = case_when(
    dx == "COVID" ~ "COVID", 
    dx == "Clear" ~ "Clear",
    lrtitype == "Viral" ~ "ViralARDS", 
    lrtitype == "Bacterial" ~ "BacterialARDS" 
  )) %>%
  mutate(dx = factor(dx, levels = c("COVID", "BacterialARDS", "ViralARDS", "Clear"))) #Set COVID as the reference value, so that all DESeq contrasts are with COVID

exp2.genes <- 
  exp1.genes %>% dplyr::select(one_of(exp2.samples$sampleid))

exp2.dds <- 
  DESeqDataSetFromMatrix(
    countData = exp2.genes,
    colData = exp2.samples,
    design = ~ dx
  )

exp2.dds <- DESeq(exp2.dds, parallel = T)

#This function extracts the results for each contrast in a DESeq object and shrinks log2-fold changes using ashr or apeglm. In addition, the function uses the IHW package to calculate false detection rates. The results for each contrast are saved in an object in the global environment. The results are also exported as a CSV file in a subfolder of the active working directory. 
deseqresults(exp2.dds, methods = "ashr")
```

### Heatmaps
The following code generates the heatmaps used in Figure 1 and Figure 2. To generate high resolution figures, the height and width of every cell is set to 10. This generates heatmaps that can only be displayed in RStudio if the screen is rotated into portrait mode (or on a very high resolution screen). On smaller screens, RStudio will generate a truncated version of image. The images should display correctly if knit to an HTML file.

```{r,fig.height = 16, fig.width = 10}
exp1.vst <- vst(exp1.dds, blind = TRUE)
exp1.nm <- exp1.vst %>% assay 
exp1.nm %<>% as.data.frame%>% rownames_to_column("ensembl_gene_id") %>% left_join(ensembltohgnc) %>% distinct(hgnc_symbol, .keep_all = T) %>% filter(hgnc_symbol != "") %>% column_to_rownames("hgnc_symbol") %>% dplyr::select(-ensembl_gene_id)

ARDSvsCOVID.DEgenes <- 
  exp1.ARDSvsCOVID.ashr %>%
  filter(ihwP < 0.1) %>% 
  mutate(direction = sign(log2FoldChange)) %>%
  group_by(direction) %>%
  arrange(pvalue, .by_group = T) %>%
  filter(row_number() <= heatmaprows) %>%
  ungroup %>%
  dplyr::select(hgnc_symbol) %>%
  unlist

ARDSvsCOVID.nm  <-  
  exp1.nm %>%
  filter(rownames(.) %in% ARDSvsCOVID.DEgenes) %>%
  t %>%
  as.data.frame %>%
  scale

exp1.breaks <- quantile_breaks(ARDSvsCOVID.nm %>% t)
exp1.annotation <- exp1.samples %>%
  dplyr::select(sampleid, dx) %>%
  column_to_rownames("sampleid")

pheatmap.rotate(
  ARDSvsCOVID.nm %>% t,
  show_rownames = T,
  show_colnames = F,
  scale = "row",
  cutree_rows = 2,
  cutree_cols = 1,
  border_color = NA,
  breaks = exp1.breaks,
  color = viridis(9),
  fontsize_col = 10,
  fontsize_row = 8,
  cluster_rows = T,
  cluster_col = T,
  treeheight_row = 20,
  treeheight_col = 20,
  annotation_legend = TRUE,
  annotation_colors = ann_colors,
  annotation_col = exp1.annotation,
  clustering_distance_cols = "euclidean",
  clustering_distance_rows = "manhattan",
  uselastdend = F,
  rotate = F, 
  cellwidth = 10,
  cellheight = 10
  )
```

```{r}
exp2.vst <- vst(exp2.dds, blind = TRUE)
exp2.nm <- exp2.vst %>% assay 
exp2.nm %<>% as.data.frame%>% rownames_to_column("ensembl_gene_id") %>% left_join(ensembltohgnc) %>% distinct(hgnc_symbol, .keep_all = T) %>% filter(hgnc_symbol != "") %>% column_to_rownames("hgnc_symbol") %>% dplyr::select(-ensembl_gene_id)
```


```{r, fig.height = 16, fig.width = 10}
BacterialARDSvsCOVID.DEgenes <- exp2.BacterialARDSvsCOVID.ashr %>%
  filter(ihwP < 0.1) %>% 
  mutate(direction = sign(log2FoldChange)) %>%
  group_by(direction) %>%
  arrange(pvalue, .by_group = T) %>%
  filter(row_number() <= heatmaprows) %>%
  ungroup %>%
  dplyr::select(hgnc_symbol) %>%
  unlist

BacterialARDSvsCOVID.samples <- 
  exp2.samples %>%
  filter(dx %in% c("COVID", "BacterialARDS")) %>% 
  dplyr::select(sampleid) %>%
  unlist

BacterialARDSvsCOVID.nm  <- 
  exp2.nm %>%
  filter(rownames(.) %in% BacterialARDSvsCOVID.DEgenes) %>%
  t %>%
  as.data.frame %>% 
  filter(rownames(.) %in% BacterialARDSvsCOVID.samples) %>%
  scale

exp2.breaks <- quantile_breaks(BacterialARDSvsCOVID.nm %>% t)
exp2.annotation <- exp2.samples %>%
  dplyr::select(sampleid, dx) %>%
  column_to_rownames("sampleid")

pheatmap.rotate(
  BacterialARDSvsCOVID.nm %>% t,
  show_rownames = T,
  show_colnames = F,
  scale = "row",
  cutree_rows = 2,
  cutree_cols = 1,
  border_color = NA,
  breaks = exp2.breaks,
  color = viridis(9),
  fontsize_col = 10,
  fontsize_row = 8,
  cluster_rows = T,
  cluster_col = T,
  treeheight_row = 20,
  treeheight_col = 20,
  annotation_legend = TRUE,
  annotation_colors = ann_colors,
  annotation_col = exp2.annotation,
  clustering_distance_cols = "euclidean",
  clustering_distance_rows = "manhattan",
  uselastdend = F,
  rotate = F, 
  cellwidth = 10,
  cellheight = 10
  )
```


```{r, fig.height = 16, fig.width = 10}
ViralARDSvsCOVID.DEgenes <- 
  exp2.ViralARDSvsCOVID.ashr %>%
  filter(ihwP < 0.1) %>% 
  mutate(direction = sign(log2FoldChange)) %>%
  group_by(direction) %>%
  arrange(pvalue, .by_group = T) %>%
  filter(row_number() <= heatmaprows) %>%
  ungroup %>%
  dplyr::select(hgnc_symbol) %>%
  unlist

ViralARDSvsCOVID.samples <- 
  exp2.samples %>%
  filter(dx %in% c("COVID", "ViralARDS")) %>% 
  dplyr::select(sampleid) %>%
  unlist

ViralARDSvsCOVID.nm  <- 
  exp2.nm %>%
  filter(rownames(.) %in% ViralARDSvsCOVID.DEgenes) %>%
  t %>%
  as.data.frame %>% 
  filter(rownames(.) %in% ViralARDSvsCOVID.samples) %>%
  scale

pheatmap.rotate(
  ViralARDSvsCOVID.nm %>% t,
  show_rownames = T,
  show_colnames = F,
  scale = "row",
  cutree_rows = 2,
  cutree_cols = 1,
  border_color = NA,
  breaks = exp2.breaks,
  color = viridis(9),
  fontsize_col = 10,
  fontsize_row = 8,
  cluster_rows = T,
  cluster_col = T,
  treeheight_row = 20,
  treeheight_col = 20,
  annotation_legend = TRUE,
  annotation_colors = ann_colors,
  annotation_col = exp2.annotation,
  clustering_distance_cols = "euclidean",
  clustering_distance_rows = "manhattan",
  uselastdend = F,
  rotate = F, 
  cellwidth = 10,
  cellheight = 10
  )
```
